impl/client/src/main/javascript/web/util/formatting.js

```diff
@@ -192,6 +192,30 @@ define("common-ui/util/formatting", ['common-ui/util/timeutil', 'common-ui/util/
           }
 
           if(timezone == 'client') {
+
+              if (value.length == 28) {
+
+                  var currentDate = new Date();
+                  var offset = -currentDate.getTimezoneOffset();
+
+                  // calculation is done in minutes because it is the smaller unit on a timezone definition
+                  // e.g.: "2018-05-20T01:12:23.456-0430" where -04h30m is the timezone setting
+                  var offsetServerMinutes = value.substring(23, 26) * 60;
+                  var diffOffset = offset - offsetServerMinutes;
+
+                  if(diffOffset != 0){
+                      this._initDateFormatters();
+
+                      var currentMinutes = ( currentDate.getHours() * 60 ) + currentDate.getMinutes();
```

Comment:
I cannot see why the number of minutes into the day is relevant...

Reply:
The number of minutes are used in order to avoid to loose precision, because timezone's granularity is in minutes (e.g. -04:30)